sdk/core/Azure.Core.TestFramework/src/ProcessTracker.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace Azure.Core.TestFramework
+{
+    // Uses Windows Job Objects to ensure external processes are killed if the current process is terminated non-gracefully.
```

Comment:
Is the expectation that recordings are made only with Windows, or is this just not a concern with non-Windows platforms i.e., non-graceful termination?

Reply:
Recordings can be made on other platforms, but this process tracker is only supported for Windows. Ideally it would work for all platforms.